src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -0,0 +1,422 @@
+    return dataset.columns[dataset.columns.duplicated()].tolist()
+
+
+def get_common_cols(staging_dataset: pd.DataFrame, main_dataset: pd.DataFrame) -> list:
+    """ Gets the list of common columns between two dataframes
+    """
+    common_cols = staging_dataset.columns.intersection(main_dataset.columns).tolist()
+    return common_cols
+
+
+def get_missing_cols(staging_dataset: pd.DataFrame, main_dataset: pd.DataFrame) -> list:
+    """ Gets the list of missing columns present in main but not in staging
+    """
+    missing_cols = main_dataset.columns.difference(staging_dataset.columns).tolist()
+    return missing_cols
+
+
+def get_additional_cols(
+    staging_dataset: pd.DataFrame, main_dataset: pd.DataFrame
+) -> list:
+    """ Gets the list of additional columns present in staging but not in main
+    """
+    add_cols = staging_dataset.columns.difference(main_dataset.columns).tolist()
+    return add_cols
+
+
+def get_S3FileSystem_from_session(
+    aws_session: boto3.session.Session,
+) -> fs.S3FileSystem:
+    """Gets a pyarrow S3 filesystem object given an
+    authenticated aws session with credentials
+
+    Args:
+        aws_session (boto3.session.Session): authenticated aws session
+
+    Returns:
+        fs.S3FileSystem: S3 filesystem object initiated from AWS credentials
+    """
+    session_credentials = aws_session.get_credentials()
+    s3_fs = fs.S3FileSystem(
+        access_key=session_credentials.access_key,
+        secret_key=session_credentials.secret_key,
+        session_token=session_credentials.token,
+    )
+    return s3_fs
+
+
+def get_parquet_dataset(
+    dataset_key: str, s3_filesystem: fs.S3FileSystem
+) -> pd.DataFrame:
+    """
+    Returns a Parquet dataset on S3 as a pandas dataframe
+
+    Args:
+        dataset_key (str): The URI of the parquet dataset.
+        s3_filesystem (S3FileSystem): A fs.S3FileSystem object
+
+    Returns:
+        pandas.DataFrame
+    """
+    table_source = dataset_key.split("s3://")[-1]
+    parquet_dataset = pq.read_table(source=table_source, filesystem=s3_filesystem)
+    return parquet_dataset.to_pandas()
+
+
+def get_folders_in_s3_bucket(
+    s3: boto3.client, bucket_name: str, namespace: str
+) -> list:
+    """Gets the folders in the S3 bucket under the specific namespace
+
+    Args:
+        s3 (boto3.client): authenticated s3 client
+        bucket_name (str): name of the S3 bucket to look into
+        namespace (str): namespace of the path to look for folders in
+
+    Returns:
+        list: folder names inside S3 bucket
+    """
+    response = s3.list_objects_v2(Bucket=bucket_name, Prefix=f"{namespace}/parquet/")
+    if "Contents" in response.keys():
+        contents = response["Contents"]
+        folders = [
+            content["Key"].split("/")[-1]
+            for content in contents
+            if content["Key"].split("/")[-1] != "owner.txt"
+        ]
+    else:
+        folders = []
+    return folders
+
+
+def keep_common_rows_cols(
```

Comment:
I'm also not sure about this method I used to get the two datasets to the same dim (combination of subsetting, merge and then separation) so we can use something like `pandas.compare`. This method is used in: `compare_dataset_row_vals`

Reply:
This will be resolved with implementation of [datacompy](https://github.com/capitalone/datacompy)